### DISCUSSION TITLE
QML support for Linux phones

### DISCUSSION THREAD

**Discussion by the-ssd** [slint-ui/slint#2805] [Ideas]
Ubuntu touch and Plasma mobile both use QML.

**Comment by tronical** [comment]
Hi! Could you elaborate what you mean with QML support in the context of Slint? Thanks:)

**Reply by the-ssd** [reply]
QML backend, like for desktop QT.
Desktop QT will work, but QML will be better.

[This is from here](https://forum.qt.io/post/437156)
QWidgets have no native GPU-support, so everything you show/animate etc. will be rendered by the CPU. For basic buttons, icons, and images that is no problem, neither for desktop pcs nor mobile devices.

GPU-supported QWidgets(e.g. QOpenGLWidget) won't work because they use OpenGL and on mobile it defaults to software rendering

QML has native GPU-Support that makes it the predestined type for "fancy" UI-s with lots of animations, slides, movies, fade ins, fade outs.

**Reply by ogoffart** [reply]
What would be the value of a QML backend? 
I would think the winit backend would work in Ubuntu touch and Plasma mobile.

Maybe you wish to use the native QML controls?  I'm afraid this might be hard as we would need to use QML to render these controls into an image or texture and then render that.  It may be easier to re-implement these controls in Slint.

**Reply by the-ssd** [reply]
If themes will work without QML, then there is none.

**Reply by the-ssd** [reply]
I wanted to write a Telegram client with slint, so it needs to look like breeze
